### ISSUE TITLE
Does MenuMeters work on the new MacBook Pro with M1 Max Chip?

### ISSUE THREAD

**Issue by kevinliu23** [yujitach/MenuMeters#257]
I can't seem to launch MenuMeters on my new MacBook Pro with the M1 Max chip. Is it supposed to work or do we need an update?

**Comment by sidaths** [comment]
Interesting, this doesn't seem to work for me on my M1 Max Macbook Pro. I can open up the application after moving it to the Applications folder, and play around with the settings, but at no point does anything show up in the menu bar. Is there some other setting I must set to get it to display

**Comment by yujitach** [comment]
@sidaths Are you using the latest one 2.1.5 ?

**Comment by sidaths** [comment]
Yes, I was using the latest version, my issue was that I had too many settings enabled which meant that nothing was displayed. It may make sense to update the defaults on install to enable fewer displays so that it shows up?

**Comment by yujitach** [comment]
Ah, I see, and that's a good suggestion. I might implement it, when I myself buy a Macbook with a notch.

In the meantime, I might suggest you to try other MenuMeters-like apps

- https://github.com/exelban/stats
- https://iglance.github.io
- https://github.com/gao-sun/eul

which would use less space on the menubar. They are better maintained and better looking, too.

**Comment by sidaths** [comment]
Thank you for the suggestions, I might give those a shot at some point. I usually only have a few items displayed and make them grayscale so that they don't stand out, and I've managed to get MenuMeters set up just fine now. Thanks for your efforts here :)

**Comment by yujitach** [comment]
You're welcome. By the way, is it that the default setting of MenuMeters is already too wide to be displayed on a Macbook with a notch, with only the system-provided things on the menubar?  In that case the default does need to be changed.

**Comment by agraf** [comment]
Yes, the default settings are already too wide for notch display and so the whole item does not get rendered. Sorry, I forgot to mention that bit in my earlier enablement patches :)

**Comment by schriftgestalt** [comment]
I tweaked it a bit that it uses much less space so it has a chance to fit
this is the network and the CPU item in a rather minimal configuration:
<img width="203" alt="Screen Shot 2021-11-01 at 17 46 10" src="https://user-images.githubusercontent.com/174660/139708454-a0567d29-1e7a-4c10-8438-5d5e0aee394b.png">

**Comment by srcshelton** [comment]
In my case (running 2.1.6.1 on a MacBook Pro 14" M1 Pro) MenuMeters doesn't appear to display anything at all - it could be disappearing under the notch, but I can't click on anything beneath it… although it's entirely possible that the OS is filtering under-notch clicks.  I also don't see anything protruding from the left-hand side: I wonder if there's some (new?) filtering preventing the display of items judged "too wide"?

**Comment by srcshelton** [comment]
From one of the other referenced options, I got this:

<img width="372" alt="Screenshot 2021-12-12 at 12 46 33 pm" src="https://user-images.githubusercontent.com/179397/145712870-ce463068-e309-4a64-8845-6ee280ffa831.png">

… so It looks like a system limitation (although it'd be nice if MenuMeters could also catch this situation and likewise give the user a clear message of what's happened).

**Comment by schriftgestalt** [comment]
Can you try to use the widget with the minimal configuration (e.g. no numbers for the CPU)

**Comment by srcshelton** [comment]
Yeah, I tried reducing the number of other menu-bar items to a practical minimum, and then set MenuMeters to display CPU cores only, and still nothing.

**Comment by schriftgestalt** [comment]
Can you try this version. It is a build from my fork?
[MenuMeters.zip](https://github.com/yujitach/MenuMeters/files/7713685/MenuMeters.zip)

**Comment by srcshelton** [comment]
Just a note to add that, as-of macOS 12.1 MenuMeters is showing up in the menu bar - but frequently gets hidden due to the notch :(

(I'm not sure if this was always the problem, or whether something's been fixed since (presumably by Apple, since I'm still on MenuMeters 2.1.6.1), but it was working before the 12.1 upgrade too, apologies for not updating this ticket!)
